metadata-db-app/src/migrations/049_add_created_at_to_collection_tables.clj

```diff
@@ -0,0 +1,57 @@
+(ns migrations.049-add-created-at-to-collection-tables
+  "Adds created_at column to collection tables."
+  (:require [clojure.java.jdbc :as j]
+            [config.migrate-config :as config]
+            [config.mdb-migrate-helper :as h]))
+
+(def revision-date-cache
+  "Cache of revision dates by concept-id to avoid repeated lookups"
+  (atom {}))
+
+(defn- add-created-at 
+  []
+  (doseq [t (h/get-collection-tablenames)]
+    (h/sql (format "alter table %s add created_at TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL" t))
+    (h/sql (format "CREATE INDEX %s_c_i ON %s(created_at)" t t))))
+
+(defn- revision-date-query-sql
+  "Returns the parameterized SQL string for getting revision dates from the given 
+  table for a concept-id"
+  [table]
+  (format "SELECT revision_date FROM %s WHERE concept_id = ? ORDER BY revision_date ASC" table))
+
+(defn- get-oldest-revision-date-for-concept-id
+  "Retrieve the oldest revision-date for the collection in the given table with the given id"
+  [sql-stmt concept-id]
+  (or (get @revision-date-cache concept-id)
+      (let [revision-date (->> (j/query (config/db) [sql-stmt concept-id])
+                               (map :revision_date)
+                               first)]
+        (swap! revision-date-cache #(assoc % concept-id revision-date))
+        revision-date)))
+       
+(defn- set-created-at
+  "Sets the value of created_at in collection tables to the earliest revision_date for each collection"
+  []
+  (doseq [t (h/get-collection-tablenames)
+          :let [rev-date-sql (revision-date-query-sql t)]
+          result (h/query (format "SELECT * from %s" t))
```

Comment:
Rather than updating the collection revisions one at a time, we should update all revisions of a collection at a time. We can do that by `select distinct concept_id from <the_table>`, and do it one collection at a time. This way, we can remove the need for `revision-date-cache` and the migration will run much faster.